assets/js/admin/wc-shipping-zone-methods.js

```diff
@@ -62,6 +62,7 @@
 							shippingMethod.trigger( 'change:methods' );
 							shippingMethod.changes = {};
 							shippingMethod.trigger( 'saved:methods' );
+							window.onbeforeunload = null;
```

Comment:
Looks like a practical fix :vulcan_salute: 

I was initially confused why we interact with the `onbeforeunload` property directly (since within this same file we set things up using an [event listener](https://github.com/woocommerce/woocommerce/pull/29510/files#diff-f6d591c256ea15681bdaec8d51ddf7937d80db8c773fd86bb185650002ddacaeR82) rather than using the window property directly), so I wonder if adding a comment might be useful for the next person to touch this, something like:

```suggestion
							// Overrides the onbeforeunload callback added by settings.js.
							window.onbeforeunload = null;
```

(Just to make it clearer that this relates specifically to the dirty page handler from that other piece of JS, and not the `setUnloadConfirmation` or `clearUnloadConfirmation` functions from this file.)

Reply:
Good idea. Added.